### ISSUE TITLE
Go to definition from template not working

### ISSUE THREAD

**Issue by anton-marchenko** [angular/vscode-ng-language-service#1414]
Go to definition not working any more (my version of extension is v12.0.5)
VS Code version: 1.57
Angular: 9.1.13
Typescript: 3.8.3

![image](https://user-images.githubusercontent.com/45929621/122171935-f1a61680-ce88-11eb-9ff6-0b90b1806dbc.png)

![image](https://user-images.githubusercontent.com/45929621/122172185-303bd100-ce89-11eb-9b87-84d0f5b4cbb2.png)

See same issue on stackoverflow: https://stackoverflow.com/questions/62950553/go-to-definition-not-working-in-vs-code-version-1-47-2

I tried to change extension version and everything works fine in v11.1.0

![image](https://user-images.githubusercontent.com/45929621/122173005-fae3b300-ce89-11eb-9e68-707eaeb8b70f.png)

**Comment by glebmachine** [comment]
Got the same!
Using with "Solution style" tsconfig.json

**Comment by atscott** [comment]
@anton-marchenko @glebmachine Can you share an example project with this issue?

**Comment by anton-marchenko** [comment]
@atscott 

Unfortunately I can not reproduce it in a simple project. In the same time I can not show our large commerce project because of NDA.

Could it be helpful if  I show you some the extension or VSCode logs without showing my code? How can I do that?

**Comment by atscott** [comment]
Let's start with this: can you share your `tsconfig.json`? Do you have `strictTemplates` enabled for your project?

Note that the Ivy based language service became the default in the v12 release. This means that information provided by the language service will exactly mirror what you've configured the compiler to understand. Previous versions independently interpreted the template, which would result in different information and diagnostics than what you would get for a build/serve.

**Comment by anton-marchenko** [comment]
@atscott 

> can you share your tsconfig.json? 

Yeah:

`{
  "compileOnSave": false,
  "compilerOptions": {
    "strict": true,
    "baseUrl": "src",
    "paths": {
      "@app/*": ["app/*"]
    },
    "outDir": "./dist/out-tsc",
    "sourceMap": true,
    "declaration": false,
    "module": "esnext",
    "moduleResolution": "node",
    "emitDecoratorMetadata": true,
    "experimentalDecorators": true,
    "target": "es5",
    "typeRoots": [
      "node_modules/@types"
    ],
    "lib": [
      "es2017",
      "dom"
    ]
  }
}`

> Do you have strictTemplates enabled for your project?

No, I don`t.

> Note that the Ivy based language service became the default in the v12 release.

There is a clue. I`ve found [here](https://stackoverflow.com/questions/65688608/angular-language-service-in-vscode-does-not-work-because-it-is-not-an-angular-pr) that I can use old engine in v12+

I enabled legacy View Engine in v12.0.5 and everything started works fine:

![image](https://user-images.githubusercontent.com/45929621/122527666-34e8bc80-d024-11eb-8147-2768b259ec7c.png)

Later I will check about strictTemplates option.

**Comment by glebmachine** [comment]
@anton-marchenko hi, comrade!)

Let's find out how to handle it together!
Could you contact me?

t.me/glebmikheev
facebook.com/glebmikheev

**Comment by atscott** [comment]
@anton-marchenko Thanks for the additional information. I created https://github.com/angular/vscode-ng-language-service/issues/1418 as a feature request to track if there is an appetite for enabling `strictTemplates`/overriding compiler options only for the purposes of the language service and not for the `ng serve`/`ng build`. 

I highly recommend enabling strict templates and fixing the errors. Those are real indications that your templates are not using the correct types in various places and could indicate that you have undetected bugs in your app. I do not recommend using legacy View Engine, as this is deprecated and will be removed in the future. Using it would be pushing the issue on to your future self for resolution.

Closing as working as intended.
